### ISSUE TITLE
Can't load data stream or unable to  access token

### ISSUE THREAD

**Issue by Resi70** [SpaceOgre/smarttv-twitch#12]
Hello,
i am able tot push this release to my UE46ES7000 tv but after opening te app no channels will load.

After load i get the error "no stream data loaded" when i do a refresh i the error "unable to  access token".

Do i have to create in this release my own API key?  I did not see anythink about this in the readme so i am wondering why its not working.

**Comment by SpaceOgre** [comment]
Which release did you download?

**Comment by Resi70** [comment]
3.9.2

**Comment by SpaceOgre** [comment]
It should work, unless Twich has banned the new Client-ID.

Please try and generate your own Api key and test.

**Comment by Fansh1ne** [comment]
3.9.2 working here. Did you enter username/userid?

**Comment by Resi70** [comment]
> 3.9.2 working here. Did you enter username/userid?

Yes i added my username in the  config.js

**Comment by Fansh1ne** [comment]
Weird. Try changing client id as mentioned above.

**Comment by Resi70** [comment]
> It should work, unless Twich has banned the new Client-ID.
> 
> Please try and generate your own Api key and test.

I can't get it to work :(

Here are the steps i followed.  I used every time the clean downloaded version.

* Deleted the not working version from my smart HUB.
* uploaded the new 3.9.2 version without editing anything. > Errors above
* Uploaded a version with my username in the config.js >same Error
* Created my own API key  in https://dev.twitch.tv/console/apps and past this id in the config.js > same error.

See screenshot. Maybe some fields have not the correct values. I did hide my api key in this image

<a href="https://ibb.co/8NHyG9x"><img src="https://i.ibb.co/8NHyG9x/twitch.jpg" alt="twitch" border="0"></a>

Any suggestions?

**Comment by SpaceOgre** [comment]
You get the error when trying to play a channel?

**Comment by Resi70** [comment]
Direct after the app has loaded i see a count from 0 tot 15 and then the error "unable to load stream data". Also when i click on A,B,C or D i get this same error. When i click on the remote on  button in the middle of the left, right, up and down arrows i get the message "unable to retrieve acces token".

**Comment by SpaceOgre** [comment]
Oh that is weird then, it seems like the app might not be able to connect to the Internet at all. The Client-ID problem should only affect playing a channel not loading the lists.

Stupid question but do you have other apps on your TV that can access Internet?

**Comment by Resi70** [comment]
Yes, i can brows internet with the browser and use the other apps that need internet like netflix and videoland.

**Comment by SpaceOgre** [comment]
Which list is showing when you start? Followed channels or Channels? With username entered in config.js

**Comment by SpaceOgre** [comment]
To know more I would need to do a testversion that prints more specific error messages. This I will not be able to do for a while because of the holidays.

What happens if you enter this url in the tvs browser: https://api.twitch.tv/helix/streams?game_id=33214
You should get a 401 error

**Comment by Resi70** [comment]
Hoping you all did have great Christmas 😀

Thank you for your suggestion👍. It is really strange. I can't visit twitch urls in my browser. Its get redirected to a local file url. If i visit https://twitch.tv i see this url in the browser.
![20191225_023618](https://user-images.githubusercontent.com/39061122/71498047-0b9a0680-285b-11ea-87ba-15f2bf5fcd03.jpg)

I did try to reset my to to factory default and even switch my smart hub to a other country and back. Still the same. I can visit other sites.

**Comment by SpaceOgre** [comment]
This is really weird, did you install the app via Sammy Widgets?

Does redirect happen in the browser even if the app Is not installed?

**Comment by Resi70** [comment]
Yes i always used sammy Widgets for installation. The problem still exists after removal from the twitch app. Even after a factory default from the tv and a reset from the smart hub. 

Seems that all outgoing connections with the twitch word in it gets blocked. Maybe a software issue.

My tv software version is 002022 and my models tv is UE46ES7000.

**Comment by SpaceOgre** [comment]
Sorry I am out of ideas, did a search for the problem but did not find any thing.

Might be hard to fix since the problems seems to be in your TVs routing. Maybe try and contact Samsung support or try their forums.

**Comment by Fansh1ne** [comment]
It might be your router blocking some connections. Try to check router firewall settings.

**Comment by Resi70** [comment]
I did some tests.  Its not my router. I connected my tv wireless to my 4g hotspot (other ISP) and still the same problem. 

Maybe it has something to do with the supported TLS version. When i visit duckduckgo.com with my tv browser i get a announcement about end of support tls version 1.0. Maybe twitch does not support this old version also. In the past everything was http now it is https.

**Comment by SpaceOgre** [comment]
Yeah that could be it, I think Twitch only support 1.2 from what I could find.
